### PR TITLE
[Fix](core) Fix wal mgr heap use after free when stop doris

### PR DESCRIPTION
**Pull request by Yukang-Lian**
Problem: When the process stops, there is a heap use after free error with the WAL manager.

Reason: During the startup process, if the storage engine does not initialize successfully and the main program directly returns 0, the WAL manager, which was created during initialization and started a thread to periodically check disk space, will encounter an issue. When the program exits and returns 0, local variables are destroyed first before the thread is properly terminated. If the thread attempts to access those local variables at this point, it leads to a heap use after free error.

Solution: Ensure that the thread for periodically checking disk space is only started after the storage engine has been successfully initialized.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by mymeiyi — APPROVED** [review]

File: be/src/olap/wal/wal_manager.cpp [hunk 1]
```diff
@@ -475,6 +475,9 @@ Status WalManager::update_wal_dir_estimated_wal_bytes(const std::string& wal_dir
 }
 
 Status WalManager::_update_wal_dir_info_thread() {
+    while (!ExecEnv::ready()) {
+        sleep(1);
```
Comment (gavinchou) [inline be/src/olap/wal/wal_manager.cpp:479]: Ensure it is 1 sec and LOG here to show what is happening.

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run p0

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run cloud_p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**